app/resources/api/v2/plate_creation_resource.rb

```diff
@@ -0,0 +1,130 @@
+    #
+    # @example POST request with arguments specified by relationship
+    #   POST /api/v2/plate_creations/
+    #   {
+    #     "data": {
+    #       "type": "plate_creations",
+    #       "attributes": {},
+    #       "relationships": {
+    #         "parent": {
+    #           "data": { "type": "plates", "id": "123" }
+    #         },
+    #         "child_purpose": {
+    #           "data": { "type": "plate_purposes", "id": "234" }
+    #         },
+    #         "user": {
+    #           "data": { "type": "users", "id": "345" }
+    #         }
+    #       }
+    #     }
+    #   }
+    #
+    # @example GET request for all PlateCreation resources
+    #   GET /api/v2/plate_creations/
+    #
+    # @example GET request for a PlateCreation with ID 123
+    #   GET /api/v2/plate_creations/123/
+    #
+    # For more information about JSON:API see the [JSON:API Specifications](https://jsonapi.org/format/)
+    # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
+    # of the JSON:API standard.
+    class PlateCreationResource < BaseResource
+      ###
+      # Attributes
+      ###
+
+      # @!attribute [w] child_purpose_uuid
+      #   This is declared for convenience where the {PlatePurpose} is not available to set as a relationship.
+      #   Setting this attribute alongside the `child_purpose` relationship will prefer the relationship value.
+      #   @deprecated Use the `child_purpose` relationship instead.
+      #   @param value [String] The UUID of the {PlatePurpose} to use when creating the child plate.
+      #   @return [Void]
+      #   @see #child_purpose
+      attribute :child_purpose_uuid, writeonly: true
+
+      def child_purpose_uuid=(value)
+        @model.child_purpose = PlatePurpose.with_uuid(value).first
+      end
+
+      # @!attribute [w] parent_uuid
+      #   This is declared for convenience where the parent {Plate} is not available to set as a relationship.
+      #   Setting this attribute alongside the `parent` relationship will prefer the relationship value.
+      #   @deprecated Use the `parent` relationship instead.
+      #   @param value [String] The UUID of the {Plate} to become the parent of the created child plate.
+      #   @return [Void]
+      #   @see #parent
+      attribute :parent_uuid, writeonly: true
+
+      def parent_uuid=(value)
+        @model.parent = Plate.with_uuid(value).first
+      end
+
+      # @!attribute [w] user_uuid
+      #   This is declared for convenience where the {User} is not available to set as a relationship.
+      #   Setting this attribute alongside the `user` relationship will prefer the relationship value.
+      #   @deprecated Use the `user` relationship instead.
+      #   @param value [String] The UUID of the {User} who initiated this plate creation.
+      #   @return [Void]
+      #   @see #user
+      attribute :user_uuid, writeonly: true
+
+      def user_uuid=(value)
```

Comment:
I wonder if this user_uuid= is something we can move to the base resource as it always has the same implementation. Might be an annoying bit of meta programming though 😆

Reply:
Yeah I think you're probably right.  Except there will be one resource that doesn't want that and then you're looking at including it as a module.  I think it's OK to get this all done with the repetition for now.  It was always repeated in v1.